test/extended/apiserver/kubeconfigs.go

```diff
@@ -34,7 +35,14 @@ var _ = g.Describe("[Conformance][sig-api-machinery][Feature:APIServer] local ku
 			})
 			o.Expect(err).NotTo(o.HaveOccurred())
 			framework.Logf("Discovered %d master nodes.", len(masterNodes.Items))
-			o.Expect(masterNodes.Items).NotTo(o.HaveLen(0))
+
+			controlPlaneTopology, err := exutil.GetControlPlaneTopology(oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// external controlplane topology doesn't have master nodes
+			if *controlPlaneTopology != configv1.ExternalTopologyMode {
```

Comment:
why not move this check to the top of the test (before trying to get the master nodes) and then Skip out of it?

1) faster skip
2) no risk of it failing due to an issue trying to get the master nodes list when you don't need it anyway
3) if someone adds more logic to this test later that also doesn't work w/ external control planes, it's a non-issue.

Reply:
(unless there is also a desire to fundamentally confirm that the 4 kubeconfigs the test client has available work correctly?)

Reply:
The reason is that we share the ExternalTopologyMode with IBM roks where it seems this test passes somehow as I can not find a skip anywhere.Hence the idea was to continue testing this even if we have an external controlplane topology if there are master nodes.

Reply:
we sure ROKS is actually running this test and not skipping it in some other way?  it is hard to understand how it would pass.

Reply:
No, not sure I just wanted to err on the side of not breaking anything. But I suppose we can change it again if there really is an issue, so moved it to the top.